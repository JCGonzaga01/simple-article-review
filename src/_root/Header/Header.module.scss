@import "helpers/stylings/mixins";
@import "helpers/stylings/global";

.wrapper {
  @include align-center;
  justify-content: space-between;
  margin-bottom: 62px;

  .adrenalin {
    height: 17px;
    width: 100px;
  }

  .menu {
    height: 20px;
    width: 20px;
  }

  .menuWrapper {
    background: rgba(0, 0, 0, 0.63);
    height: 100vh;
    margin: 0 -20px;
    position: fixed;
    top: 0;
    width: 100vw;
    z-index: 2;

    .menuContainer {
      align-items: flex-end;
      animation: showing 0.3s ease-in;
      background: rgba(0, 0, 0, 0.72);
      display: flex;
      flex-direction: column;
      height: 100vh;
      padding-top: 40px;
      position: fixed;
      right: 0;
      top: 0;
      width: 200px;
      z-index: 1;

      .closeIcon {
        cursor: pointer;
        height: 25px;
        margin: 0 20px 20px 0;
        width: 25px;
      }
    }
  }

  .headerLinks {
    @include media-query(sp) {
      @include align-center;
      flex-direction: column;
      width: 100%;
    }

    color: $black;
    font-family: $fontType1;
    font-size: $fontXS1;
    font-weight: 400;
    line-height: $lineheightXS1;

    & > span {
      @include media-query(sp) {
        color: $white;
        margin-bottom: 20px;
      }

      &:hover {
        @include media-query(sp) {
          background: $white;
          color: $black;
          padding: 0 5px;
        }
        cursor: pointer;
        background: $black;
        border-radius: 3px;
        color: $white;
      }
      &:not(:last-child) {
        @include media-query(sp) {
          margin-right: 0;
        }
        margin-right: 33px;
      }
    }
  }
}

@keyframes showing {
  0% {
    opacity: 0;
    transform: translateX(100%);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}
