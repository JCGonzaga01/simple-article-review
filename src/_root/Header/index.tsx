import React, { useState, useEffect } from "react";
import { headerLinks } from "constants/global";
import styles from "./Header.module.scss";

const Header: React.FC = () => {
  const [isSPView, setIsSPView] = useState(false);
  const [isOpenMenu, setIsOpenMenu] = useState(false);

  useEffect(() => {
    if (!window.matchMedia) return;
    const mediaQuerySP = window.matchMedia("(max-width: 743px)");
    const updateMediaQuery = () => setIsSPView(mediaQuerySP.matches);
    updateMediaQuery();

    mediaQuerySP.addListener(updateMediaQuery);
    return () => mediaQuerySP.removeListener(updateMediaQuery);
  }, []);

  const handleOnClickMenu = () => setIsOpenMenu(!isOpenMenu);

  return (
    <div className={styles.wrapper}>
      <img
        className={styles.adrenalin}
        src={process.env.PUBLIC_URL + "/assets/adrenalin.svg"}
        alt={"adrenalin"}
      />
      {isSPView ? (
        !isOpenMenu && (
          <img
            className={styles.menu}
            src={process.env.PUBLIC_URL + "/assets/menu.svg"}
            alt={"Menu"}
            onClick={handleOnClickMenu}
          />
        )
      ) : (
        <div className={styles.headerLinks}>
          {headerLinks.map((text, idx) => (
            <span key={idx}>{text}</span>
          ))}
        </div>
      )}
      {isOpenMenu && (
        <div className={styles.menuWrapper}>
          <div className={styles.menuContainer}>
            <img
              className={styles.closeIcon}
              src={process.env.PUBLIC_URL + "/assets/close.png"}
              alt={"close"}
              onClick={handleOnClickMenu}
            />
            <div className={styles.headerLinks}>
              {headerLinks.map((text, idx) => (
                <span key={idx} onClick={handleOnClickMenu}>
                  {text}
                </span>
              ))}
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default Header;
