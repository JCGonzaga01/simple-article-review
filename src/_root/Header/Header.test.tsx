import React from 'react';
import Header from './index';
import renderer from 'react-test-renderer';

describe('<Header />', () => {
  let wrapper: any;
  let instance: any;

  beforeEach(() => {
    wrapper = renderer.create(<Header />);
    instance = wrapper.root;
  });

  test('should mount component', () => {
    const component = renderer.create(
      <Header />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display header links", () => {
    expect(instance.findByProps({className: 'headerLinks'}).children).toHaveLength(6)
  });
});
