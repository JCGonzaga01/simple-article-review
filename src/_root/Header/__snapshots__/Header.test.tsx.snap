// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should mount component 1`] = `
<div
  className="wrapper"
>
  <img
    alt="adrenalin"
    className="adrenalin"
    src="/assets/adrenalin.svg"
  />
  <div
    className="headerLinks"
  >
    <span>
      Culture
    </span>
    <span>
      Work
    </span>
    <span>
      Clients
    </span>
    <span>
      Services
    </span>
    <span>
      Careers
    </span>
    <span>
      Contact
    </span>
  </div>
</div>
`;
